### ISSUE TITLE
Seeing inference time increase 10-15% with  FP16 for MaskRCNN. 

### ISSUE THREAD

**Issue by whatdhack** [NVIDIA/TensorRT#500]
## Description

With trtserver, observing 10-15% increase in inference time with FP16  optimization enabled. It was suugested in the triton-inference-server list that I contact TensorRT list about the issue.  Trtserver  config and log follows. 

## Environment

Using nvcr.io/nvidia/tritonserver:20.03-py3

**TensorRT Version**: 
**GPU Type**:  T4
**Nvidia Driver Version**:  440.33.01
**CUDA Version**: 10.2
**CUDNN Version**: 
**Operating System + Version**:  Ubuntu 18.04
**Python Version (if applicable)**: 
**TensorFlow Version (if applicable)**: 
**PyTorch Version (if applicable)**: 
**Baremetal or Container (if container which image + tag)**: 

## Relevant Files
trtserver config.pbtxt:
```
name: "maskrcnn"
platform: "tensorflow_savedmodel"
max_batch_size: 0
input [
        {
        name: "image:0"
        data_type: TYPE_FP32
        dims:  -1
        dims:  -1
        dims:   3
      }
]
output [
        {
        name: "output/boxes:0"
        data_type: TYPE_FP32
        dims: -1
        dims:  4
      },
      {
        name: "output/scores:0"
        data_type: TYPE_FP32
        dims: -1
      },
      {
        name: "output/labels:0"
        data_type: TYPE_INT64
        dims: -1
      }
]
optimization
{ execution_accelerators {
   gpu_execution_accelerator : [ {
     name : "tensorrt"
     parameters { key: "precision_mode" value: "FP16" }}]
}}
```

Logs:
```
2020-04-15 06:44:04.113328: I tensorflow/compiler/tf2tensorrt/segment/segment.cc:460] There are 1070 ops of 6 different types in the graph that are not converted to TensorRT: Assi

gn, RestoreV2, PlaceholderWithDefault, Const, NoOp, VariableV2, (For more information see https://docs.nvidia.com/deeplearning/frameworks/tf-trt-user-guide/index.html#supported-op

s).
2020-04-15 06:44:04.113758: I tensorflow/compiler/tf2tensorrt/convert/convert_graph.cc:633] Number of TensorRT candidate segments: 0
2020-04-15 06:44:04.785002: I tensorflow/cc/saved_model/loader.cc:332] SavedModel load for tags { serve }; Status: success. Took 17675592 microseconds.
I0415 06:44:04.785522 943 dynamic_batch_scheduler.cc:233] Starting dynamic-batch scheduler thread 0 at nice 5...
I0415 06:44:04.785565 943 base_backend.cc:127] backend for XXXXXXXX
.
.
.
x, Where, MatMul, CropAndResize, GatherNd, Fill, Greater, Sub, Tile, Cast, TopKV2, NonMaxSuppressionV3, Identity, VariableV2, ReverseV2, ExpandDims, FusedBatchNorm, MirrorPad, GatherV2, NoOp, ConcatV2, Placeholder, Minimum, StridedSlice, Reshape, Add, Squeeze, BiasAdd, Shape, Pack, Transpose, RealDiv, Slice, Conv2D, Size, (For more information see https://docs.nvidia.com/deeplearning/frameworks/tf-trt-user-guide/index.html#supported-ops).
2020-04-16 16:20:55.091690: I tensorflow/compiler/tf2tensorrt/convert/convert_graph.cc:734] TensorRT node TRTEngineOp_0 added for segment 0 consisting of 30 nodes succeeded.
2020-04-16 16:20:55.091836: I tensorflow/compiler/tf2tensorrt/convert/convert_graph.cc:734] TensorRT node TRTEngineOp_2 added for segment 2 consisting of 8 nodes succeeded.
2020-04-16 16:20:55.091954: I tensorflow/compiler/tf2tensorrt/convert/convert_graph.cc:734] TensorRT node decode_bbox_target/TRTEngineOp_4 added for segment 4 consisting of 22 nodes succeeded.
2020-04-16 16:20:55.092023: I tensorflow/compiler/tf2tensorrt/convert/convert_graph.cc:734] TensorRT node decode_bbox_target_1/TRTEngineOp_5 added for segment 5 consisting of 22 nodes succeeded.
2020-04-16 16:20:55.092170: I tensorflow/compiler/tf2tensorrt/convert/convert_graph.cc:734] TensorRT node TRTEngineOp_6 added for segment 6 consisting of 10 nodes succeeded.
ed.
ed.
2020-04-16 16:20:55.092336: I tensorflow/compiler/tf2tensorrt/convert/convert_graph.cc:734] TensorRT node group3/block0/TRTEngineOp_9 added for segment 9 consisting of 5 nodes succeeded.
2020-04-16 16:20:55.092385: I tensorflow/compiler/tf2tensorrt/convert/convert_graph.cc:734] TensorRT node TRTEngineOp_10 added for segment 10 consisting of 7 nodes succeeded.
2020-04-16 16:20:55.092434: I tensorflow/compiler/tf2tensorrt/convert/convert_graph.cc:734] TensorRT node TRTEngineOp_11 added for segment 11 consisting of 6 nodes succeeded.
gment 12 consisting of 4 nodes succeeded.
d on function objects.
2020-04-16 16:20:55.109708: W tensorflow/compiler/tf2tensorrt/convert/trt_optimization_pass.cc:183] TensorRTOptimizer is probably called on funcdef! This optimizer must *NOT* be called on function objects.
2020-04-16 16:20:55.110594: W tensorflow/compiler/tf2tensorrt/convert/trt_optimization_pass.cc:183] TensorRTOptimizer is probably called on funcdef! This optimizer must *NOT* be called on function objects.
2020-04-16 16:20:55.112029: W tensorflow/compiler/tf2tensorrt/convert/trt_optimization_pass.cc:183] TensorRTOptimizer is probably called on funcdef! This optimizer must *NOT* be called on function objects.
2020-04-16 16:20:55.112930: W tensorflow/compiler/tf2tensorrt/convert/trt_optimization_pass.cc:183] TensorRTOptimizer is probably called on funcdef! This optimizer must *NOT* be called on function objects.
2020-04-16 16:20:55.113733: W tensorflow/compiler/tf2tensorrt/convert/trt_optimization_pass.cc:183] TensorRTOptimizer is probably called on funcdef! This optimizer must *NOT* be called on function objects.
2020-04-16 16:20:55.114452: W tensorflow/compiler/tf2tensorrt/convert/trt_optimization_pass.cc:183] TensorRTOptimizer is probably called on funcdef! This optimizer must *NOT* be called on function objects.
2020-04-16 16:20:55.115815: W tensorflow/compiler/tf2tensorrt/convert/trt_optimization_pass.cc:183] TensorRTOptimizer is probably called on funcdef! This optimizer must *NOT* be called on function objects.
2020-04-16 16:20:55.116887: W tensorflow/compiler/tf2tensorrt/convert/trt_optimization_pass.cc:183] TensorRTOptimizer is probably called on funcdef! This optimizer must *NOT* be called on function objects.
2020-04-16 16:20:55.118755: W tensorflow/compiler/tf2tensorrt/convert/trt_optimization_pass.cc:183] TensorRTOptimizer is probably called on funcdef! This optimizer must *NOT* be called on function objects.
2020-04-16 16:20:55.119809: W tensorflow/compiler/tf2tensorrt/convert/trt_optimization_pass.cc:183] TensorRTOptimizer is probably called on funcdef! This optimizer must *NOT* be called on function objects.
2020-04-16 16:20:55.120499: W tensorflow/compiler/tf2tensorrt/convert/trt_optimization_pass.cc:183] TensorRTOptimizer is probably called on funcdef! This optimizer must *NOT* be called on function objects.
2020-04-16 16:20:55.121172: W tensorflow/compiler/tf2tensorrt/convert/trt_optimization_pass.cc:183] TensorRTOptimizer is probably called on funcdef! This optimizer must *NOT* be called on function objects.
2020-04-16 16:20:55.402649: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcublas.so.10
2020-04-16 16:20:55.644534: I tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op.cc:733] Building a new TensorRT engine for TRTEngineOp_2 input shapes: [[1,960,540,3]]
2020-04-16 16:20:55.644622: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libnvinfer.so.7
2020-04-16 16:20:55.645130: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libnvinfer_plugin.so.7
2020-04-16 16:20:55.645130: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libnvinfer_plugin.so.7
2020-04-16 16:20:57.004705: W tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:37] DefaultLogger Current optimization profile is: 0. Please ensure there are no enqueued operations pending in this context prior to switching profiles
2020-04-16 16:20:57.005034: I tensorflow/stream_executor/platform/default/dso_loader.cc:44] Successfully opened dynamic library libcudnn.so.7
2020-04-16 16:20:57.305480: I tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op.cc:733] Building a new TensorRT engine for TRTEngineOp_3 input shapes: [[1,64,480,270]]
@@@
2020-04-16 16:20:57.895732: W tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:37] DefaultLogger Current optimization profile is: 0. Please ensure there are no enqueued operations pending in this context prior to switching profiles
2020-04-16 16:20:58.340652: I tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op.cc:733] Building a new TensorRT engine for group1/block0/TRTEngineOp_7 input shapes: [[1,128,240,135]]
2020-04-16 16:20:58.405655: W tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:37] DefaultLogger Current optimization profile is: 0. Please ensure there are no enqueued operations pending in this context prior to switching profiles
2020-04-16 16:20:58.699315: I tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op.cc:733] Building a new TensorRT engine for group2/block0/TRTEngineOp_8 input shapes: [[1,256,120,67]]
2020-04-16 16:20:58.757496: W tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:37] DefaultLogger Current optimization profile is: 0. Please ensure there are no enqueued operations pending in this context prior to switching profiles
2020-04-16 16:20:59.408881: I tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op.cc:733] Building a new TensorRT engine for decode_bbox_target/TRTEngineOp_4 input shapes: [[29700,2,2], [29700,2,2]]
@@@
2020-04-16 16:20:59.543261: W tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:37] DefaultLogger Current optimization profile is: 0. Please ensure there are no enqueued operations pending in this context prior to switching profiles
2020-04-16 16:20:59.544875: I tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op.cc:733] Building a new TensorRT engine for TRTEngineOp_0 input shapes: [[1000,4]]
I0416 16:20:59.550411 199 server.cc:224] Polling model repository
2020-04-16 16:20:59.750188: W tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:37] DefaultLogger Current optimization profile is: 0. Please ensure there are no enqueued operations pending in this context prior to switching profiles
2020-04-16 16:20:59.750575: I tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op.cc:733] Building a new TensorRT engine for roi_align/crop_and_resize/transform_fpcoor_for_tf/TRTEngineOp_12 input shapes: [[1000,1], [1000,1], [1000,1], [1000,1]]
@@@
2020-04-16 16:20:59.779093: W tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:37] DefaultLogger Current optimization profile is: 0. Please ensure there are no enqueued operations pending in this context prior to switching profiles
2020-04-16 16:20:59.779381: I tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op.cc:733] Building a new TensorRT engine for TRTEngineOp_10 input shapes: [[1000,28,28,1024]]
I0416 16:21:14.550869 199 server.cc:224] Polling model repository
I0416 16:21:29.551246 199 server.cc:224] Polling model repository
2020-04-16 16:21:33.189718: W tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:37] DefaultLogger Current optimization profile is: 0. Please ensure there are no enqueued operations pending in this context prior to switching profiles
2020-04-16 16:21:34.870265: I tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op.cc:733] Building a new TensorRT engine for group3/block0/TRTEngineOp_9 input shapes: [[1000,512,14,14]]
2020-04-16 16:21:35.235690: W tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:37] DefaultLogger Current optimization profile is: 0. Please ensure there are no enqueued operations pending in this context prior to switching profiles
2020-04-16 16:21:39.709996: I tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op.cc:733] Building a new TensorRT engine for TRTEngineOp_6 input shapes: [[1000,2048,7,7], [1000,2048,7,7]]
2020-04-16 16:21:41.255460: W tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:37] DefaultLogger Current optimization profile is: 0. Please ensure there are no enqueued operations pending in this context prior to switching profiles
2020-04-16 16:21:41.255819: I tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op.cc:733] Building a new TensorRT engine for TRTEngineOp_1 input shapes: [[1000,56]]
2020-04-16 16:21:41.255903: I tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op.cc:733] Building a new TensorRT engine for TRTEngineOp_11 input shapes: [[1000,14]]
2020-04-16 16:21:41.299566: W tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:37] DefaultLogger Current optimization profile is: 0. Please ensure there are no enqueued operations pending in this context prior to switching profiles
2020-04-16 16:21:41.301971: W tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:37] DefaultLogger Current optimization profile is: 0. Please ensure there are no enqueued operations pending in this context prior to switching profiles
2020-04-16 16:21:41.302086: I tensorflow/compiler/tf2tensorrt/kernels/trt_engine_op.cc:733] Building a new TensorRT engine for decode_bbox_target_1/TRTEngineOp_5 input shapes: [[14000,2,2], [14000,2,2]]
2020-04-16 16:21:41.437978: W tensorflow/compiler/tf2tensorrt/utils/trt_logger.cc:37] DefaultLogger Current optimization profile is: 0. Please ensure there are no enqueued operations pending in this context prior to switching profiles
I0416 16:21:41.438855 199 base_backend.cc:573] output 'output/boxes:0' is GPU tensor: 0
I0416 16:21:41.438892 199 http_server.cc:434] HTTP allocation: 'output/boxes:0', size: 1248, addr: 0x7ff391a0d9c0
I0416 16:21:41.438909 199 base_backend.cc:573] output 'output/labels:0' is GPU tensor: 0
I0416 16:21:41.438923 199 http_server.cc:434] HTTP allocation: 'output/labels:0', size: 624, addr: 0x7ff4a26fe7f0
I0416 16:21:41.438931 199 base_backend.cc:573] output 'output/scores:0' is GPU tensor: 0
I0416 16:21:41.438941 199 http_server.cc:434] HTTP allocation: 'output/scores:0', size: 312, addr: 0x7ff4a27092b0
I0416 16:21:41.439635 199 http_server.cc:445] HTTP release: size 1248, addr 0x7ff391a0d9c0
I0416 16:21:41.439649 199 http_server.cc:445] HTTP release: size 624, addr 0x7ff4a26fe7f0
I0416 16:21:41.439654 199 http_server.cc:445] HTTP release: size 312, addr 0x7ff4a27092b0
I0416 16:21:41.450240 199 http_server.cc:456] HTTP request: 2 /api/infer/xxxx
I0416 16:21:41.450305 199 model_repository_manager.cc:511] GetInferenceBackend() 'xxxx' version -1
I0416 16:21:41.450374 199 base_backend.cc:643] Running xxxx_0_gpu0 with 1 request payloads
I0416 16:21:41.450481 199 base_backend.cc:455] input 'image:0' is GPU tensor: 0
I0416 16:21:44.454070 199 base_backend.cc:573] output 'output/boxes:0' is GPU tensor: 0
I0416 16:21:44.454109 199 http_server.cc:434] HTTP allocation: 'output/boxes:0', size: 1248, addr: 0x7ff391a0d9c0
I0416 16:21:44.454119 199 base_backend.cc:573] output 'output/labels:0' is GPU tensor: 0
I0416 16:21:44.454125 199 http_server.cc:434] HTTP allocation: 'output/labels:0', size: 624, addr: 0x7ff4a26fd870
I0416 16:21:44.454132 199 base_backend.cc:573] output 'output/scores:0' is GPU tensor: 0
I0416 16:21:44.454142 199 http_server.cc:434] HTTP allocation: 'output/scores:0', size: 312, addr: 0x7ff4a26f7e70
I0416 16:21:44.454242 199 http_server.cc:445] HTTP release: size 1248, addr 0x7ff391a0d9c0
I0416 16:21:44.454253 199 http_server.cc:445] HTTP release: size 624, addr 0x7ff4a26fd870
I0416 16:21:44.454257 199 http_server.cc:445] HTTP release: size 312, addr 0x7ff4a26f7e70
I0416 16:21:44.462411 199 http_server.cc:456] HTTP request: 2 /api/infer/xxxx
I0416 16:21:44.462473 199 model_repository_manager.cc:511] GetInferenceBackend() 'xxxx' version -1
I0416 16:21:44.462526 199 base_backend.cc:643] Running xxxx_0_gpu0 with 1 request payloads

```

**Comment by rajeevsrao** [comment]
@whatdhack can you please check if this is resolved for you in 20.09 (or 20.10 to be posted on 10.23)?

**Comment by ttyio** [comment]
I will close since get no response for more than 3 weeks, please reopen if you still have question, thanks!
